plugins/carla/cmake/carla-bridge-native.cmake

```diff
@@ -0,0 +1,56 @@
+add_executable(carla-bridge-native)
+mark_as_advanced(carla-bridge-native)
+
+target_compile_definitions(
+  carla-bridge-native
+  PRIVATE BUILDING_CARLA BUILD_BRIDGE BUILD_BRIDGE_ALTERNATIVE_ARCH CARLA_BACKEND_NAMESPACE=CarlaOBS
+          CARLA_LIB_EXT="${CMAKE_SHARED_LIBRARY_SUFFIX}" $<$<BOOL:${X11_FOUND}>:HAVE_X11>)
+
+target_compile_options(carla-bridge-native PRIVATE $<$<BOOL:${MSVC}>:/wd4244 /wd4267 /wd4273>
+                                                   $<$<NOT:$<BOOL:${MSVC}>>:-Wno-error -Werror=vla> ${X11_CFLAGS})
+
+target_include_directories(
+  carla-bridge-native
+  PRIVATE carla/source
+          carla/source/backend
+          carla/source/backend/engine
+          carla/source/backend/plugin
+          carla/source/includes
+          carla/source/modules
+          carla/source/utils
+          ${X11_INCLUDE_DIRS})
+
+target_link_directories(carla-bridge-native PRIVATE ${X11_LIBRARY_DIRS})
```

Comment:
Do not use CMake 2.x-style variables to link targets - use a proper `X11` find module to yield targets of the libraries you need. Those will carry the library names, as well as include directories, library directories, and other usage requirements (which are automatically added to your target by CMake).

Reply:
while this works for X11 because it has a cmake file, it wont work for pkg-config style dependencies.
I am hoping to introduce them later for a separate PR, but I guess it can be a problem not dealt with now.  Will adjust for reducing complexity of the first PR.